main/options.c

```diff
@@ -2081,10 +2081,10 @@ static booleanOption BooleanOptions [] = {
  *  Generic option parsing
  */
 
-static void checkOptionOrder (const char* const option)
+static void checkOptionOrder (const char* const option, boolean more_hyphen)
```

Comment:
Could I ask you to rename the argument to `longOption`?
Makes it more clear what TRUE means in this case.